src/hooks/useSpaceParams.ts

```diff
@@ -0,0 +1,20 @@
+import { useParams } from "react-router-dom";
+import { useSearchParam } from "react-use";
+
+export interface SpaceParams {
+  spaceSlug?: string;
+}
+
+/**
+ * Retrieve the venueId from the URL path (/:spaceSlug/) or the search query string (?spaceSlug=).
+ *
+ * @see https://reactrouter.com/web/api/Hooks/useparams
+ */
+export const useSpaceParams: () => string | undefined = () => {
```

Comment:
This can also be one in some next PR. 

A rename:
```suggestion
export const useSpaceSlug: () => string | undefined = () => {
```

or a retype:
```suggestion
export const useSpaceParams: () => { spaceSlug: string | undefined } = () => {
```